tasks/sep/006-1-sc-changes/SignFromJson.s.sol

```diff
@@ -0,0 +1,523 @@
+    function checkOptimismMintableERC20Factory() internal view {
+        console.log("Running assertions on the OptimismMintableERC20Factory");
+
+        require(proxies.OptimismMintableERC20Factory.code.length != 0, "4800");
+
+        require(EIP1967Helper.getImplementation(proxies.OptimismMintableERC20Factory).code.length != 0, "4801");
+
+        OptimismMintableERC20Factory factoryToCheck = OptimismMintableERC20Factory(proxies.OptimismMintableERC20Factory);
+
+        require(factoryToCheck.BRIDGE() == proxies.L1StandardBridge, "4900");
+        require(factoryToCheck.bridge() == proxies.L1StandardBridge, "5000");
+        require(factoryToCheck.bridge().code.length != 0, "5001");
+        require(EIP1967Helper.getImplementation(factoryToCheck.bridge()).code.length != 0, "5002");
+    }
+
+    /// @notice Asserts that the L1ERC721Bridge is setup correctly
+    function checkL1ERC721Bridge() internal view {
+        console.log("Running assertions on the L1ERC721Bridge");
+
+        require(proxies.L1ERC721Bridge.code.length != 0, "5100");
+
+        require(EIP1967Helper.getImplementation(proxies.L1ERC721Bridge).code.length != 0, "5101");
+
+        L1ERC721Bridge bridgeToCheck = L1ERC721Bridge(proxies.L1ERC721Bridge);
+
+        require(address(bridgeToCheck.OTHER_BRIDGE()) == Predeploys.L2_ERC721_BRIDGE, "5200");
+        require(address(bridgeToCheck.otherBridge()) == Predeploys.L2_ERC721_BRIDGE, "5300");
+
+        require(address(bridgeToCheck.MESSENGER()) == proxies.L1CrossDomainMessenger, "5400");
+        require(address(bridgeToCheck.messenger()) == proxies.L1CrossDomainMessenger, "5500");
+        require(address(bridgeToCheck.messenger()).code.length != 0, "5502");
+
+        require(address(bridgeToCheck.superchainConfig()) == proxies.SuperchainConfig, "5600");
+        require(address(bridgeToCheck.superchainConfig()).code.length != 0, "5601");
+        require(EIP1967Helper.getImplementation(address(bridgeToCheck.superchainConfig())).code.length != 0, "5603");
+    }
+
+    /// @notice Asserts the OptimismPortal is setup correctly
+    function checkOptimismPortal() internal view {
+        console.log("Running assertions on the OptimismPortal");
+
+        require(proxies.OptimismPortal.code.length != 0, "5700");
+
+        require(EIP1967Helper.getImplementation(proxies.OptimismPortal).code.length != 0, "5701");
+
+        OptimismPortal portalToCheck = OptimismPortal(payable(proxies.OptimismPortal));
+
+        require(address(portalToCheck.systemConfig()) == proxies.SystemConfig, "6000");
+        require(address(portalToCheck.systemConfig()) == proxies.SystemConfig, "6100");
+        require(address(portalToCheck.systemConfig()).code.length != 0, "6101");
+        require(EIP1967Helper.getImplementation(address(portalToCheck.systemConfig())).code.length != 0, "6102");
+
+        require(portalToCheck.GUARDIAN() == superchainConfigGuardian, "6200");
```

Comment:
```suggestion
        require(portalToCheck.guardian() == superchainConfigGuardian, "6200");
```

Reply:
Should we just remove all of the IMMUTABLE getters?

Reply:
yea I think that's ideal, the only ones I see left are the L2OO ones but there's no L2OO with FP on sepolia anyway so I'm surprised that one's passing

Reply:
Well, it's still there, right?

Reply:
Seems it's because the L2OO address still is in the registry, so even though the proxy isn't connected it's still checked

Reply:
A bit weird that none of the getters on the portal started failing though, there should be one for the oracle

Reply:
I don't see a call to the oracle getter on the portal in `checkOptimismPortal`. Looks like we have a `systemConfig.l2OutputOracle()` call, but I don't think the SystemConfig was updated by FP, though seems like it should have been